### PR TITLE
Methods adds

### PR DESCRIPTION
**Pull request by AndreyGalkin**

### CODE REVIEW

File: saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java [hunk 1]
```diff
@@ -319,6 +351,7 @@ private boolean sendGETRequest(String servicePath,
         if (params != null) {
             url = url.concat(SETools.paramsToString(params));
         }
+        Log.v("tag", "url " + url);
```
Comment (deltadd) [inline saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java]: log

File: saltedge-library/src/main/java/com/saltedge/sdk/utils/SEConstants.java [hunk 2]
```diff
@@ -134,7 +134,7 @@ of this software and associated documentation files (the "Software"), to deal
     public static final String KEY_RETURN_LOGIN_ID = "return_login_id";
     public static final String KEY_PROVIDER_MODES = "provider_modes";
     public static final String KEY_FROM_ID = "from_id";
-    public static final String KEY_FROM_DATE = "from_id";
+    public static final String KEY_FROM_DATE = "from_date";
     public static final String KEY_TO_DATE = "from_id";
```
Comment (deltadd) [inline saltedge-library/src/main/java/com/saltedge/sdk/utils/SEConstants.java]: something wrong

File: saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java [hunk 3]
```diff
@@ -21,6 +21,8 @@ of this software and associated documentation files (the "Software"), to deal
 */
 package com.saltedge.sdk.network;
 
+import android.util.Log;
```
Comment (ConstantinKV) [inline saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java]: remove

File: saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java [hunk 4]
```diff
@@ -129,12 +133,25 @@ public void onSuccessResponse(int statusCode, JSONObject response) {
 /**
  * Provider list
  * */
-    public void listingProviders(boolean clearArray, FetchListener listener) {
+
+    public void listingProviders(FetchListener listener) {
```
Comment (ConstantinKV) [inline saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java]: fetchProviders

File: saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java [hunk 5]
```diff
@@ -151,9 +166,10 @@ public void onSuccessResponse(int statusCode, JSONObject response) {
                 }
                 nextPageId = paginationAvailable(response);
```
Comment (ConstantinKV) [inline saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java]: rename method

File: saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java [hunk 6]
```diff
@@ -149,11 +164,13 @@ public void onSuccessResponse(int statusCode, JSONObject response) {
                 for (SEProvider entry : gson.fromJson(providerArrayString, SEProvider[].class)) {
                     providerArray.add(entry);
                 }
-                nextPageId = paginationAvailable(response);
+
+                String pageId = getNextPageId(response);
+                nextPageId = !pageId.equals(nextPageId) ? pageId : "";
```
Comment (ConstantinKV) [inline saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java]: switch condition

File: saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java [hunk 7]
```diff
@@ -129,12 +131,25 @@ public void onSuccessResponse(int statusCode, JSONObject response) {
...
+    }
+
+    private void providerList(final Date date, FetchListener listener) {
         fetchListener = listener;
-        if (providerArray == null || providerArray.isEmpty() || clearArray) {
+        HashMap<String, String> params = new HashMap<>();
+        if (date != null) {
+            params.put(SEConstants.KEY_FROM_DATE, SEDateTools.parseDateToShortString(date));
+        }
+        if (providerArray == null || providerArray.isEmpty() || nextPageId.isEmpty()) {
```
Comment (ConstantinKV) [inline saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java]: providerArray.isEmpty() condition useless

File: saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java [hunk 8]
```diff
@@ -149,11 +164,13 @@ public void onSuccessResponse(int statusCode, JSONObject response) {
                 for (SEProvider entry : gson.fromJson(providerArrayString, SEProvider[].class)) {
```
Comment (ConstantinKV) [inline saltedge-library/src/main/java/com/saltedge/sdk/network/SERequestManager.java]: parse whole array and use addAll

**Pull request merged**